modules/apps/transfer/keeper/migrations.go

```diff
@@ -43,6 +43,11 @@ func (m Migrator) MigrateTraces(ctx sdk.Context) error {
 			if !equalTraces(newTrace, dt) {
 				newTraces = append(newTraces, newTrace)
 			}
+
+			if !m.keeper.bankKeeper.HasDenomMetaData(ctx, newTrace.IBCDenom()) {
+				m.keeper.SetDenomMetaData(ctx, newTrace)
+			}
+
```

Comment:
This migration may have already been run and might not be run again. I think it is safer to make a separate migration only setting the denom metadata. Please let me know if you have any questions on how to do this. I'm also happy to receive that change in a followup pr, it would make reviews faster/easier

Reply:
I have created a new migration and registered it with as version 3 migration. Feel free to take up the task if required.